src/invoke_training/lora/utils.py

```diff
@@ -0,0 +1,125 @@
+import typing
+
+import torch
+
+from invoke_training.lora.layers import BaseLoRALayer
+from invoke_training.lora.lora_block import LoRABlock
+
+
+def find_modules(
```

Comment:
How do you plan on deciding which device to use with torch? In the main app we do a lot of passing around a torch.device object between functions. Could be nice if there's a cleaner way.

Reply:
Good call out. I hadn't thought about that yet here. As a start, I think I'll update `from_layer(...)` to copy the device from the original layer. I can't think of a use case for wanting to override that, but it would be easy enough to add a `device` parameter to `inject_lora_layers(...)` in the future if we wanted to.

I'll make this change in PR #2

Reply:
could be a fast follow in a new PR to get everything consolidated into main for now

Reply:
Addressed in 2225965